### PR TITLE
remove Channel type

### PR DESCRIPTION
**Pull request by ruanxin**
Related issue: https://github.com/kyma-project/lifecycle-manager/issues/324

### CODE REVIEW

File: api/v1alpha1/kyma_types.go [hunk 1]
```diff
@@ -97,7 +98,8 @@ type Sync struct {
 // KymaSpec defines the desired state of Kyma.
 type KymaSpec struct {
 	// Channel specifies the desired Channel of the Installation, usually targeting different module versions.
-	Channel Channel `json:"channel"`
+	// +kubebuilder:validation:Pattern:=^[a-z]+$
+	Channel string `json:"channel"`
```
Comment (Tomasz-Smelcerz-SAP) [inline api/v1alpha1/kyma_types.go:102]: Validation as above

File: api/v1alpha1/kyma_types.go [hunk 2]
```diff
@@ -139,26 +141,10 @@ type KymaStatus struct {
 
 	// Active Channel
 	// +optional
-	ActiveChannel Channel `json:"activeChannel,omitempty"`
+	ActiveChannel string `json:"activeChannel,omitempty"`
```
Comment (Tomasz-Smelcerz-SAP) [inline api/v1alpha1/kyma_types.go:144]: I assume this should have the same validation as the "Channel" field.

File: api/v1alpha1/moduletemplate_types.go [hunk 3]
```diff
@@ -35,7 +35,8 @@ import (
 type ModuleTemplateSpec struct {
 	// Channel is the targeted channel of the ModuleTemplate. It will be used to directly assign a Template
 	// to a target channel. It has to be provided at any given time.
-	Channel Channel `json:"channel"`
+	// +kubebuilder:validation:Pattern:=^[a-z]+$
+	Channel string `json:"channel"`
```
Comment (Tomasz-Smelcerz-SAP) [inline api/v1alpha1/moduletemplate_types.go:39]: Validation as above

**Review by Tomasz-Smelcerz-SAP — CHANGES_REQUESTED** [review]
I think we must set a min and max for channel length, to avoid single-letter channel names and more than 10k chars long names. The latter could be used for DOS attacks.

**Review by Tomasz-Smelcerz-SAP — COMMENTED** [review]
I think we must set a min and max for channel length, to avoid single-letter channel names and names with more than, say, 10k chars. The latter could be used for DOS attacks.

**Review by adityabhatia — APPROVED** [review]

**Pull request merged**